drivers/sensor/apds9253/apds9253.c

```diff
@@ -121,9 +121,115 @@ static int apds9253_channel_get(const struct device *dev, enum sensor_channel ch
 	return 0;
 }
 
+static int apds9253_attr_set_gain(const struct device *dev, enum sensor_gain_apds9253 gain)
+{
+	const struct apds9253_config *config = dev->config;
+	struct apds9253_data *drv_data = dev->data;
+	uint8_t value;
+
+	if (drv_data->gain == gain) {
+		return 0;
+	}
+
+	switch (gain) {
+	case APDS9253_SENSOR_GAIN_1:
+		value = APDS9253_LS_GAIN_RANGE_1;
+		break;
+	case APDS9253_SENSOR_GAIN_3:
+		value = APDS9253_LS_GAIN_RANGE_3;
+		break;
+	case APDS9253_SENSOR_GAIN_6:
+		value = APDS9253_LS_GAIN_RANGE_6;
+		break;
+	case APDS9253_SENSOR_GAIN_9:
+		value = APDS9253_LS_GAIN_RANGE_9;
+		break;
+	case APDS9253_SENSOR_GAIN_18:
+		value = APDS9253_LS_GAIN_RANGE_18;
+		break;
+	default:
+		return -EINVAL;
+	}
```

Comment:
`APDS9253_SENSOR_GAIN_1` through `APDS9253_SENSOR_GAIN_18` are sequential enums, it'll be better (to avoid branching) to just have a static lookup table:

```
static int apds9253_attr_set_gain(const struct device *dev, enum sensor_gain_apds9253 gain)
{
  static uint8_t value_map = {
      APDS9253_LS_GAIN_RANGE_1,
      APDS9253_LS_GAIN_RANGE_3,
      APDS9253_LS_GAIN_RANGE_6,
      APDS9253_LS_GAIN_RANGE_9,
      APDS9253_LS_GAIN_RANGE_18,
  };

  if (gain < APDS9253_LS_GAIN_RANGE_1 || gain > APDS9253_LS_GAIN_RANGE_18) {
    return -EINVAL;
  }
  value = value_map[gain];
```